### PR TITLE
Invoice item methods

### PR DESCRIPTION
**Pull request by jrwhitmer**
In this PR we have added all of the invoice_item methods and invoice_item_repo methods

### CODE REVIEW

**Review by rcasias — APPROVED** [review]
Everything look great to me here! Great job Jacq!

File: lib/invoice_item_repo.rb [hunk 1]
```diff
@@ -24,4 +24,50 @@ def invoice_item_instances(csv_files)
...
+
+  def update(id, attributes)
+    invoice_item = find_by_id(id)
+    if !invoice_item.nil?
+      invoice_item.quantity = attributes[:quantity] unless attributes[:quantity].nil?
+      invoice_item.unit_price = BigDecimal(attributes[:unit_price], 5) unless attributes[:unit_price].nil?
+      invoice_item.updated_at = Time.now
+    end
+    invoice_item
+  end
```
Comment (deebot10) [inline lib/invoice_item_repo.rb:60]: Nice, Methods look great for invoice_item

**Review by deebot10 — APPROVED** [review]

File: lib/invoice_item_repo.rb [hunk 2]
```diff
@@ -24,4 +24,50 @@ def invoice_item_instances(csv_files)
...
+  def invoice_item_exists?(id)
+    invoice_item = find_by_id(id)
+    invoice_item != nil
+  end
+
+  def delete(id)
+    invoice_item = find_by_id(id)
+    if invoice_item_exists?(id)
+      @invoice_item_list.delete(invoice_item)
+    end
```
Comment (ahenzler) [inline lib/invoice_item_repo.rb:71]: these methods look clean and great

File: spec/invoice_item_repo_spec.rb [hunk 3]
```diff
@@ -22,5 +22,53 @@
       expect(invoice_item_repo.invoice_item_list[0]).to be_instance_of(InvoiceItem)
       expect(invoice_item_repo.invoice_item_list.length).to eq(21830)
     end
+
+    it 'can return all invoice items' do
+      expect(invoice_item_repo.all.length).to eq(21830)
```
[the remark this answers is not in the document]
Comment (ahenzler) [reply spec/invoice_item_repo_spec.rb:27]: do we need to implement the _ for decimal marks?

**Review by ahenzler — APPROVED** [review]

**Pull request merged**